portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,21 +40,28 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
     response.getWriter().println(convertToJsonUsingGson(comments));
+    
   }
 
   private String convertToJsonUsingGson(ArrayList<String> jsonData) {
     return gson.toJson(jsonData);
   }
   
-  // Get the comments from the user and display them
+  // Get the comments from the user and add them to datastore
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String input = getParameter(request, "text-input", "");
     comments.add(input);
 
-    
-    response.setContentType("text/html");
-    response.getWriter().println(comments);
+    Entity commentEntity = new Entity("Comments");
+    commentEntity.setProperty("Comment", input);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
What's the query for here? It doesn't look like you're doing anything with the results of this query.

Reply:
The query isn't doing anything yet, I'm supposed to use it in the next step

Reply:
With the changes I just pushed, when I try to get the comments to show up on the html page, all i get is a single quotation mark (") for each comment that is entered during that session. The datastore is working properly because when I check it all of the comments  from past sessions and current session are displayed. Can you help me see where I'm going wrong with this?

Reply:
You're querying for "Task" instead of COMMENT. Also, I recommend moving your query to doGet as that's where you want to have the most up-to-date data.

Reply:
Ok thank you, that worked. I wasnt 100% clear that Queries were used to fetch the data from the database but that makes sense now